x-pack/plugins/enterprise_search/server/lib/ml/ml_model_deployment_common.ts

```diff
@@ -5,13 +5,15 @@
  * 2.0.
  */
 
+import { ELASTIC_MODEL_DEFINITIONS } from '@kbn/ml-trained-models-utils';
+
 import {
   ElasticsearchResponseError,
   isNotFoundException,
   isResourceNotFoundException,
 } from '../../utils/identify_exceptions';
 
-export const acceptableModelNames = ['.elser_model_1', '.elser_model_1_SNAPSHOT'];
+export const acceptableModelNames = Object.keys(ELASTIC_MODEL_DEFINITIONS);
```

Comment:
I'm not sure if you need that. The method provided by the `ml` plugin is taking care of the validation.

Reply:
I would like to hear @demjened thoughts because he is the original author of this piece of code.

Reply:
This variable is used in `throwIfNotAcceptableModelName()` below, which runs as a sanity check when the user presses on the "Deploy ELSER" (= download) or "Start ELSER" (= deploy once it's downloaded) buttons. There is an API behind these actions which take a model ID. The goal is not to validate the model ID, but to limit the actions to the ELSER model.

For example the download action that calls `trainedModelsProvider.putTrainedModel()` should not be executable with anything but ELSER, since we [create the request body in the code](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/server/lib/ml/start_ml_model_download.ts#L52-L62), and this particular structure might not apply to other models.

That said, these APIs are only invoked from Kibana, and if a user passed a different model ID, that would be and edge case. So if you think this check is unnecessary, feel free to remove it.